manifests/params.pp

```diff
@@ -74,7 +74,11 @@
   $agent                     = true
 
   # Custom puppetmaster
-  $puppetmaster              = $::puppetmaster
+  if $::trusted and $::trusted['authenticated'] == 'local' {
```

Comment:
Here's a weird observation - if you enable strict_variables but not trusted_node_data then Puppet throws an error saying ::trusted is undefined.  It seems you need to do `if defined('$trusted') and $::trusted['authenticated'] == 'local'` instead. (The lack of "::" in the defined call is deliberate for it to work.)